test/DynamicFees.t.sol

```diff
@@ -146,17 +148,186 @@ contract TestDynamicFees is Test, Deployers, GasSnapshot {
+        assertEq(manager.protocolFeesAccrued(currency0), expectedProtocolFee);
+
+        assertEq(_fetchPoolLPFee(key), 999999);
+    }
+
+    function test_swap_100PercentFee_AmountIn_WithProtocol() public {
+        assertEq(_fetchPoolLPFee(key), 0);
+
+        dynamicFeesHooks.setFee(1000000);
+
+        vm.prank(address(feeController));
+        manager.setProtocolFee(key, 1000);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -1000, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        vm.expectEmit(true, true, true, true, address(manager));
+        emit Swap(key.toId(), address(swapRouter), -1000, 0, SQRT_RATIO_1_1, 1e18, -1, 1000000);
+
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+
+        uint256 expectedProtocolFee = uint256(-params.amountSpecified) * 1000 / 1e6;
+        assertEq(manager.protocolFeesAccrued(currency0), expectedProtocolFee);
+    }
+
+    function test_emitsSwapFee() public {
+        assertEq(_fetchPoolLPFee(key), 0);
+
+        dynamicFeesHooks.setFee(123);
+
+        vm.prank(address(feeController));
+        manager.setProtocolFee(key, 1000);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        vm.expectEmit(true, true, true, true, address(manager));
+        emit Swap(key.toId(), address(swapRouter), -100, 98, 79228162514264329749955861424, 1e18, -1, 1122);
+
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+
+        assertEq(_fetchPoolLPFee(key), 123);
+    }
+
+    function test_fuzz_ProtocolAndLPFee(uint24 lpFee, uint16 protocolFee0, uint16 protocolFee1, int256 amountSpecified)
+        public
+    {
+        assertEq(_fetchPoolLPFee(key), 0);
+
+        lpFee = uint16(bound(lpFee, 0, 1000000));
+        protocolFee0 = uint16(bound(protocolFee0, 0, 1000));
+        protocolFee1 = uint16(bound(protocolFee1, 0, 1000));
+        vm.assume(amountSpecified != 0);
+
+        uint24 protocolFee = (uint24(protocolFee1) << 12) | uint24(protocolFee0);
+        dynamicFeesHooks.setFee(lpFee);
+
+        vm.prank(address(feeController));
+        manager.setProtocolFee(key, protocolFee);
+
+        IPoolManager.SwapParams memory params = IPoolManager.SwapParams({
+            zeroForOne: true,
+            amountSpecified: amountSpecified,
+            sqrtPriceLimitX96: SQRT_RATIO_1_2
+        });
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        snapStart("swap with dynamic fee and protocol fee");
```

Comment:
this snapshot should be in a different test because itll just be writing every fuzz run and leaving us with the final run which might change. better to do a gas test of some "normal" values to do the snapshot and then do fuzz without gas snaps

Reply:
oh okay nice, I just removed it since we already have "swap with lp fee and protocol fee"